resources/aws/subnet.go

```diff
@@ -0,0 +1,141 @@
+package aws
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	microerror "github.com/giantswarm/microkit/error"
+
+	awsclient "github.com/giantswarm/aws-operator/client/aws"
+)
+
+type Subnet struct {
+	AvailabilityZone string
+	CidrBlock        string
+	Name             string
+	VpcID            string
+	id               string
+	AWSEntity
+}
+
+func (s *Subnet) findExisting() (*ec2.Subnet, error) {
```

Comment:
This doesn't need to be have a pointer receiver. WDYT, should we pay attention to these things?